src/lib/dnssd/Resolver_ImplMinimalMdns.cpp

```diff
@@ -341,6 +342,9 @@ void PacketDataReporter::OnComplete(ActiveResolveAttempts & activeAttempts)
         mNodeData.LogNodeIdResolved();
         mNodeData.PrioritizeAddresses();
 
+        const System::Clock::Timestamp currentTime = System::SystemClock().GetMonotonicTimestamp();
+        mNodeData.mExpiryTime                      = currentTime + System::Clock::Seconds16(kDefaultTtlSeconds);
```

Comment:
@cecille instead of stuffing a default, do you know where can I get the TTL from?

Reply:
It should be parsed:

https://github.com/project-chip/connectedhomeip/blob/master/src/lib/dnssd/minimal_mdns/Parser.h#L83

Reply:
During parsing, when implementing `void OnResource(ResourceType type, const ResourceData & data) override;` the data will have a GetTtlSeconds() that you can store. We could pick the ttl of one of the A/AAAA records or the srv or the smallest of all expiries.

Reply:
Spoke with Jerry offline - let's do this in two parts. This is still better than what we have now, and setting this to the default will unblock fixing the cirque flakes. I will change this to be the correct TTL in a followup PR. Jerry will link the issue.

Reply:
Have filed #15489 to track the real fix, as well as left a comment.